src/app/shared/entry/url-deconstruct.pipe.ts

```diff
@@ -0,0 +1,17 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@ Pipe({
+  name: 'urlDeconstruct'
+})
+export class UrlDeconstructPipe implements PipeTransform {
+
+  transform(providerUrl: string, versionName: string): any {
+    if (providerUrl.includes('github.com') && versionName !== '') {
+      return (new URL(providerUrl).pathname).substr(1) + '/tree/' + versionName;
```

Comment:
If not a valid URL, e.g., does not start with http(s), this code  and the other instance below will probably throw an exception.

Not sure if it's worth worrying about, but just mentioning.

Reply:
Good point. I'm already mutating the string URL pathname in the return statements. I will try moving away from the URL library in favor of working directly on the providerUrl string